pyproj/_crs.pyx

```diff
@@ -391,15 +500,56 @@ cdef class _CRS:
         """
         Returns
         -------
-        pyproj.CRS: The datum as a CRS.
+        Datum: The datum.
         """
-        cdef PJ *projobj = proj_crs_get_datum(self.projctx, self.projobj)
+        if self._datum is not None:
+            return self._datum
+        self._datum = Datum.create(self.projctx, self.projobj)
+        return self._datum
+
+    @property
+    def source_crs(self):
+        """
+        Returns
+        -------
+        CRS: The source CRS.
+        """
+        if self._source_crs is not None:
+            return self._source_crs
+        cdef PJ * projobj
+        projobj = proj_get_source_crs(self.projctx, self.projobj)
         if projobj == NULL:
             return None
         try:
-            return self.__class__(_to_wkt(self.projctx, projobj))
+            self._source_crs = self.__class__(_to_wkt(self.projctx, projobj))
         finally:
             proj_destroy(projobj) # deallocate temp proj
+        return self._source_crs
+
+    @property
+    def sub_crs_list(self):
+        """
+        If the CRS is a compound CRS, it will return a list of sub CRS objects.
+
+        Returns
+        -------
+        list[CRS]
+
+        """
+        if self._sub_crs_list is not None:
+            return self._sub_crs_list
+        cdef int iii = 0
+        cdef PJ * projobj = proj_crs_get_sub_crs(self.projctx, self.projobj, iii)
+        self._sub_crs_list = []
+        while projobj != NULL:
+            try:
+                self._sub_crs_list.append(self.__class__(_to_wkt(self.projctx, projobj)))
+            finally:
+                proj_destroy(projobj) # deallocate temp proj
+            iii += 1
+            projobj = proj_crs_get_sub_crs(self.projctx, self.projobj, iii)
```

Comment:
It might be that this only can be 0 and 1 (for horizontal and vertical crs), so that this extra complexity of the while loop is not needed (at least the compound CRS defined in the database are like that)

Reply:
That would be nice if that can be verified. I would be surprised if it was not the case, but always good to check.

Reply:
So as already mentioned, in the database there are only CompoundCRS pre-defined with 2 components. But checking the proj.4 source code: in the CompoundCRS class itself, it also uses loops to handle the components. So at least in theory you can create a CompoundCRS with more than 2 component CRSses, from C++ to be clear. In the C api however, the creation method for a CompoundCRS just takes the horizontal and vertical CRS. 
But given it is like that in the C++ code, let's leave it here as a loop as well.

Reply:
:+1: Thanks for checking!